### PR TITLE
feature(db_reader_thread): Make process of it

### PR DESCRIPTION
**Pull request by dkropachev**
Log reading is triggering GIL, we need to move it out to a separate process

Here is logs rate and sct log processing latency correlation chart:
sct-screw-avg

Same graph but reduced by top 10 peaks:
sct-screw-cleared

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
First run gave following results

Here is logs rate and sct log processing latency correlation chart:
sct-fixed-screw

Same graph but reduced by top 10 peaks:
sct-fixed-screw-cleaned

**Comment by dkropachev** [comment]
So, avarage latency in normal operational state was reduced from ~5 to ~0.5, which is great.
But we still have kind of corelation between `number of log records` and `sct screw` source of which is unknown.

Also looking at peaks I found that some of them related to `db node restart`, probably `syslog-ng restriming logs` from file buffer. 
But some of them happened in between of the nemesis and have realy unknwon nature.

**Comment by dkropachev** [comment]
New run fix all fixes:

sct-fixed-screw-2

Cleaned from 2 peaks:
sct-fixed-screw-2-cleaned

File: sdcm/sct_events/events_device.py [hunk 1]
```diff
@@ -60,7 +61,7 @@ def __init__(self, _registry: EventsProcessesRegistry):
         self._sub_port = multiprocessing.Value(ctypes.c_uint16, 0)
         self._queue = multiprocessing.Queue()
         self._raw_events_lock = multiprocessing.RLock()
-
+        self._is_alive_flag = multiprocessing.Semaphore()
```
Comment (enaydanov) [inline sdcm/sct_events/events_device.py:64]: Why do we need another attribute which says us that a device is running? Is it not enough to check `self._running`?

Comment (dkropachev) [reply sdcm/sct_events/events_device.py]: You are right I have overlooked it.

**Review by enaydanov — APPROVED** [review]

**Comment by dkropachev** [comment]
Testing on r5.large gave following picture:

sct-screw-r5 large

Here is what was before:

sct-screw-t3 large

**Comment by dkropachev** [comment]
So there is no virtually improvements there, which means it is not a CPU that does that.

**Review by bentsi — APPROVED** [review]

**Pull request merged**